eZ/Publish/API/Repository/Tests/FieldType/FileSearchBaseIntegrationTest.php

```diff
@@ -0,0 +1,205 @@
+
+    /**
+     * If storage data should not be cleaned up
+     *
+     * @var boolean
+     */
+    protected static $leaveStorageData = false;
+
+    /**
+     * List of path in config:storage_dir that must not be deleted, and must be ignored in these tests
+     * Workaround for FieldType vs. Repository tests. The repository tests require those files since they
+     * match the ones referenced in the database fixtures used by Services tests.
+     * @var array
+     */
+    protected static $ignoredPathList;
+
+    /**
+     * Returns the install dir used by the test
+     *
+     * @return string
+     */
+    protected function getInstallDir()
+    {
+        return self::$installDir;
+    }
+
+    /**
+     * Returns the storage dir used by the test
+     *
+     * @return string
+     */
+    protected function getStorageDir()
+    {
+        return self::$storageDir;
+    }
+
+    /**
+     * Perform storage directory setup on first execution
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        if ( !isset( self::$installDir ) )
+        {
+            self::$installDir = $this->getConfigValue( 'ezpublish.kernel.root_dir' );
+            self::$storageDir = $this->getConfigValue( static::$storageDirConfigKey );
+            self::$ioRootDir = $this->getConfigValue( 'io_root_dir' );
+
+            self::setUpIgnoredPath( $this->getConfigValue( 'ignored_storage_files' ) );
+        }
+    }
+
+    /**
+     * Tears down the test.
+     *
+     * Cleans up the storage directory, if it was used
+     *
+     * @return void
+     */
+    public static function tearDownAfterClass()
+    {
+        parent::tearDownAfterClass();
+        self::cleanupStorageDir();
+    }
+
+    /**
+     * Returns an iterator over the full storage dir.
+     *
+     * @return Iterator
+     */
+    protected static function getStorageDirIterator()
+    {
+        return new RecursiveIteratorIterator(
+            new RecursiveDirectoryIterator(
+                self::$installDir . '/' . self::$storageDir,
+                FileSystemIterator::KEY_AS_PATHNAME | FileSystemIterator::SKIP_DOTS | FilesystemIterator::CURRENT_AS_FILEINFO
+            ),
+            RecursiveIteratorIterator::CHILD_FIRST
+        );
+    }
+
+    /**
+     * Removes the given directory path recursively
+     *
+     * @param string $dir
+     *
+     * @return void
+     */
+    protected static function cleanupStorageDir()
+    {
+        if ( self::$installDir == null || self::$storageDir == null || self::$leaveStorageData )
+        {
+            // Nothing to do
+            return;
+        }
+
+        try
+        {
+            $iterator = self::getStorageDirIterator();
+
+            foreach ( $iterator as $path => $fileInfo )
+            {
+                if ( $fileInfo->isDir() )
+                {
+                    if ( !self::isIgnoredPath( dirname( $path ) ) )
```

Comment:
Hi, this part was actually copied from `FileSearchBaseIntegrationTest`, it is here temporarily until we implement search tests for all file based field types.